src/libinjector/linux/linux_utils.h

```diff
@@ -0,0 +1,112 @@
+#ifndef LINUX_UTILS_H
+#define LINUX_UTILS_H
+
+#include <errno.h>
+#include <glib.h>
+#include <inttypes.h>
+#include <json-c/json.h>
+#include <libinjector/libinjector.h>
+#include <libvmi/libvmi.h>
+#include <libvmi/x86.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <assert.h>
+#include <libdrakvuf/libdrakvuf.h>
+#include <libinjector/private.h>
+
+typedef enum
+{
+    INJECT_RESULT_SUCCESS,
+    INJECT_RESULT_TIMEOUT,
+    INJECT_RESULT_CRASH,
+    INJECT_RESULT_ERROR_CODE,
+    INJECT_RESULT_METHOD_UNSUPPORTED,
+} inject_result_t;
+
+typedef enum
+{
+    STEP1,
+    STEP2,
+    STEP3,
+    STEP4,
+    STEP5,
+    STEP6,
+    STEP7,
+    STEP8,
+    STEP9,
+} injector_step_t;
+
+typedef enum
+{
+    sys_read = 0,
+    sys_write = 1,
+    sys_open = 2,
+    sys_close = 3,
+    sys_stat = 4,
+    sys_mmap = 9,
+    sys_mprotect = 10,
+    sys_munmap = 11,
+    sys_exit = 60,
+    sys_kill = 62,
+} syscall_t;
```

Comment:
I think that this is already defined in libc as SYS_exit, SYS_kill, SYS_read, etc.

Reply:
I couldn't find it, could you please tell me the header file to include?

Reply:
I think that keeping syscall numbers in the source is OK. Technically SYS_* constants refer to the system running drakvuf, which in theory could be different from the guest, e.g. 32bit host vs 64 bit guest.
I'd keep the enum ;)

Reply:
Ok